examples/passing/ContextSimplification.purs

```diff
@@ -0,0 +1,13 @@
+module Main where
+
+import Prelude
+import Control.Monad.Eff.Console
+
+shout = log <<< (<> "!") <<< show
+
+-- Here, we should simplify the context so that only one Show
+-- constraint is added.
```

Comment:
No, it only tests that _if_ simplification happens, then the generated code works. This is less trivial than it sounds since simplification involves moving type class dictionaries around in the generated code.